idaes/models/properties/general_helmholtz/helmholtz_functions.py

```diff
@@ -563,12 +782,22 @@ def basic_calculations(
 
     def s(self, **kwargs):
         """Mixed phase entropy"""
-        blk, delta_liq, delta_vap, tau, x, c = self.basic_calculations(**kwargs)
-        self.add_funcs(names=["s_func"])
-        s = (
-            blk.s_func(c, delta_liq, tau, _data_dir) * (1 - x)
-            + blk.s_func(c, delta_vap, tau, _data_dir) * x
-        )
+        sv, blk, c, u, p = self._state_vars(**kwargs)
```

Comment:
It bugs me that the first state variable is always assigned to `u`, even though it could be any one of internal energy, entropy, or enthalpy. I think confusion around this point was why I gave up reviewing this PR the first time. It's doesn't affect the code's functionality, but it's bad for the maintainability to use confusing naming conventions.

Reply:
Noted.  I'm planing to follow this up soon that just uses direct property calls for everything without going through delta and tau.  I'll change it then.